### DISCUSSION TITLE
Help Needed: Using GeoParquet, FlatGeobuf, and PMTiles with deck.gl and maplibre

### DISCUSSION THREAD

**Discussion by yharby** [visgl/loaders.gl#2910] [Q&A]
Hi,

I found some examples on loaders.gl for using GeoParquet, FlatGeobuf, and PMTiles. Here are the links:

- GeoParquet: https://loaders.gl/examples/geoparquet
- FlatGeobuf: https://loaders.gl/examples/flatgeobuf
- PMTiles: https://loaders.gl/examples/pmtiles
- GeoArrow ... GeoPackge ..etc

I'm trying to use these with deck.gl but I'm finding it hard to understand how to do it. The example codes are confusing for me.

Can anyone help me understand how to use these with deck.gl (getting started tutorial)? I'd really appreciate any advice or tips.

Thanks!

**Comment by ibgreen** [comment]
https://deck.gl/docs/developer-guide/loading-data#support-additional-formats

**Reply by ibgreen** [reply]
Not that PMTiles is not a one-shot loader like the other formats, it is based a new `TileSource` concept that is not yet supported by deck itself, so unfortunately you would need to follow the existing example or use the undocumented `TileSourceLayer` from the `@deck.gl-community/layers` module.

**Reply by kbatchu** [reply]
Is the source code available somewhere for the above mentioned examples? It is not clear to me as well.
Thanks.

**Reply by ibgreen** [reply]
The code is in 
- https://github.com/visgl/loaders.gl/tree/master/examples/website/geospatial
- https://github.com/visgl/loaders.gl/tree/master/examples/website/tiles
